File: Rust/src/beggars.rs
// https://www.codewars.com/kata/59590976838112bfea0000fa/rust

pub fn beggars(values: &[u32], n: usize) -> Vec<u32>
{
	let mut arr: Vec<u32> = vec![];
	for i in 0..n
	{
		let mut res = 0;
		for j in (i..values.len()).step_by(n)
		{
			res += values[j];
		}
		arr.push(res.to_owned());
	}
	return arr;
}

File: C++/src/Beggars.cpp
// https://www.codewars.com/kata/59590976838112bfea0000fa/cpp

#include "Beggars.hpp"

std::vector<int> Beggars::Get(const std::vector<int>& values, unsigned int n)
{
	auto arr = std::vector<int>(n);
	for (unsigned int i = 0; i < n; ++i)
	{
		int res = 0;
		for (unsigned int j = i; j < values.size(); j += n)
		{
			res += values[j];
		}
		arr[i] = res;
	}
	return arr;
}
